packages/react/src/contexts/ClerkContextProvider.tsx

```diff
@@ -146,9 +146,9 @@ function deriveState(
       lastOrganizationMember: null,
     };
   }
-  const userId: string | null | undefined = state.user ? state.user.id : state.user;
+  const userId: string | null | undefined = state.user ? state.user.id : (state.user as null | undefined);
```

Comment:
❓ Is this type assertion required?

Reply:
Without this, `npm test` will give me this issue:

```
src/contexts/ClerkContextProvider.tsx:149:9 - error TS2322: Type 'string | UserResource' is not assignable to type 'string'.
      Type 'UserResource' is not assignable to type 'string'.

    149   const userId: string | null | undefined = state.user ? state.user.id : state.user;
```

Alternatively these also work:

```ts
const userId: string | null | undefined = state.user && state.user.id;
```